### DISCUSSION TITLE
Console Spam, preventing command execution

### DISCUSSION THREAD

**Discussion by AwesomeSam676** [Pryaxis/TShock#2820] [Q&A]
Hello, I'm Sam. (AwesomeSam)

I am fairly new to using TShock, and I'm using Ubuntu v22.04.1 and .NET 6 to host version 5.0.0.0 successfully. I've done all my configs successfully (seen below in case if you see that something's wrong), but there's a lot of console spam when I attempt to play in the server. 
First off, I notice I can't move NPC housing without understanding as to why, and I can't do anything in the console. I have a few IPs to ban, since there are many people who mess my servers up. I can't do that exactly, as shown here.

![image](https://user-images.githubusercontent.com/95329261/200989628-551b6b05-a0b5-4ae7-8a6c-27c6654d5d8f.png)

I can't mainly because it's very slow and becomes unresponsive, and doesn't even show the command that I am typing. This mayhem is only caused when someone joins the server, and then problems are caused. Until then, everything works fine.

`{
  "Settings": {
    "ServerPassword": "",
    "ServerPort": 7777,
    "MaxSlots": 10,
    "ReservedSlots": 5,
    "ServerName": "world",
    "UseServerName": true,
    "LogPath": "tshock/logs",
    "DebugLogs": false,
    "DisableLoginBeforeJoin": false,
    "IgnoreChestStacksOnLoad": false,
    "AutoSave": true,
    "AnnounceSave": true,
    "ShowBackupAutosaveMessages": true,
    "BackupInterval": 10,
    "BackupKeepFor": 240,
    "SaveWorldOnCrash": true,
    "SaveWorldOnLastPlayerExit": true,
    "InvasionMultiplier": 2,
    "DefaultMaximumSpawns": 20,
    "DefaultSpawnRate": 600,
    "InfiniteInvasion": false,
    "PvPMode": "always",
    "SpawnProtection": false,
    "SpawnProtectionRadius": 5,
    "RangeChecks": true,
    "HardcoreOnly": false,
    "MediumcoreOnly": true,
    "SoftcoreOnly": false,
    "DisableBuild": false,
    "DisableHardmode": false,
    "DisableDungeonGuardian": false,
    "DisableClownBombs": false,
    "DisableSnowBalls": false,
    "DisableTombstones": false,
    "DisablePrimeBombs": false,
    "ForceTime": "normal",
    "DisableInvisPvP": false,
    "MaxRangeForDisabled": 10,
    "RegionProtectChests": false,
    "RegionProtectGemLocks": true,
    "IgnoreProjUpdate": false,
    "IgnoreProjKill": false,
    "AllowCutTilesAndBreakables": false,
    "AllowIce": false,
    "AllowCrimsonCreep": true,
    "AllowCorruptionCreep": true,
    "AllowHallowCreep": true,
    "StatueSpawn200": 3,
    "StatueSpawn600": 6,
    "StatueSpawnWorld": 10,
    "PreventBannedItemSpawn": true,
    "PreventDeadModification": true,
    "PreventInvalidPlaceStyle": true,
    "ForceXmas": false,
    "ForceHalloween": false,
    "AllowAllowedGroupsToSpawnBannedItems": false,
    "RespawnSeconds": 10,
    "RespawnBossSeconds": 20,
    "AnonymousBossInvasions": true,
    "MaxHP": 500,
    "MaxMP": 200,
    "BombExplosionRadius": 7,
    "GiveItemsDirectly": false,
    "DefaultRegistrationGroupName": "default",
    "DefaultGuestGroupName": "guest",
    "RememberLeavePos": true,
    "MaximumLoginAttempts": 3,
    "KickOnMediumcoreDeath": false,
    "MediumcoreKickReason": "Death results in a kick",
    "BanOnMediumcoreDeath": false,
    "MediumcoreBanReason": "Death results in a ban",
    "DisableDefaultIPBan": false,
    "EnableWhitelist": false,
    "WhitelistKickReason": "You are not on the whitelist.",
    "ServerFullReason": "Server is full",
    "ServerFullNoReservedReason": "Server is full without any reserved slots open.",
    "KickOnHardcoreDeath": false,
    "HardcoreKickReason": "Death results in a kick",
    "BanOnHardcoreDeath": false,
    "HardcoreBanReason": "Death results in a ban",
    "KickProxyUsers": true,
    "RequireLogin": true,
    "AllowLoginAnyUsername": false,
    "AllowRegisterAnyUsername": true,
    "MinimumPasswordLength": 4,
    "BCryptWorkFactor": 7,
    "DisableUUIDLogin": false,
    "KickEmptyUUID": false,
    "TilePaintThreshold": 15,
    "KickOnTilePaintThresholdBroken": false,
    "MaxDamage": 1175,
    "MaxProjDamage": 1175,
    "KickOnDamageThresholdBroken": false,
    "TileKillThreshold": 60,
    "KickOnTileKillThresholdBroken": false,
    "TilePlaceThreshold": 32,
    "KickOnTilePlaceThresholdBroken": false,
    "TileLiquidThreshold": 50,
    "KickOnTileLiquidThresholdBroken": false,
    "ProjIgnoreShrapnel": true,
    "ProjectileThreshold": 50,
    "KickOnProjectileThresholdBroken": false,
    "HealOtherThreshold": 50,
    "KickOnHealOtherThresholdBroken": false,
    "SuppressPermissionFailureNotices": false,
    "DisableModifiedZenith": false,
    "DisableCustomDeathMessages": true,
    "CommandSpecifier": "/",
    "CommandSilentSpecifier": ".",
    "DisableSpewLogs": false,
    "DisableSecondUpdateLogs": false,
    "SuperAdminChatRGB": [
      245,
      78,
      56
    ],
    "SuperAdminChatPrefix": "[Admin] ",
    "SuperAdminChatSuffix": "",
    "EnableGeoIP": false,
    "DisplayIPToAdmins": true,
    "ChatFormat": "{1}{2}{3}: {4}",
    "ChatAboveHeadsFormat": "{2}",
    "EnableChatAboveHeads": true,
    "BroadcastRGB": [
      127,
      255,
      212
    ],
    "StorageType": "sqlite",
    "SqliteDBPath": "tshock.sqlite",
    "MySqlHost": "localhost:3306",
    "MySqlDbName": "",
    "MySqlUsername": "",
    "MySqlPassword": "",
    "UseSqlLogs": false,
    "RevertToTextLogsOnSqlFailures": 10,
    "RestApiEnabled": false,
    "RestApiPort": 7878,
    "LogRest": false,
    "EnableTokenEndpointAuthentication": false,
    "RESTMaximumRequestsPerInterval": 5,
    "RESTRequestBucketDecreaseIntervalMinutes": 1,
    "ApplicationRestTokens": {}
  }
}`

**Comment by sgkoishi** [comment]
Console spam is known to be caused by the terminal: some terminal does not recognize the "set title" action and they print the console title out. However, you can still enter commands and execute them (but both input and output are messed with the spam).
For moving NPC, [this permission](https://github.com/Pryaxis/TShock/blob/general-devel/docs/permission-descriptions.md#tshockworldmovenpc) might be what you want.

**Reply by AwesomeSam676** [reply]
So, is there a fix for the console spam?

**Reply by sgkoishi** [reply]
https://github.com/Pryaxis/TShock/blob/f7553fde83f859d6a3f9db131fbf35a6d1358267/TShockAPI/Utils.cs#L1062-L1070

It is not officially fixed, but you can simply add a `return;` between L1065 and L1066: the GitHub Actions will build a version for you (in your fork).

**Comment by pcider** [comment]
If you're using `screen` in detached mode, you could try using the flag `-T xterm-256color`. Apparently this issue happens in detached mode (so far I've only confirmed this on my machine)

**Reply by Ortintzki** [reply]
Confirmed that this works for me as well.  It resolved all the terminal corruption I encountered when starting my tshock server.

`screen -T xterm-256color -dmS terraria ./TShock.Server`

**Reply by Ortintzki** [reply]
Sorry, using Ubuntu 20.04 LTS and TShock 5.1.3 for Terraria 1.4.4.9
